src/storage/storage.cpp

```diff
@@ -184,643 +182,670 @@ Storage::ReturnCode Storage::Run(int max_wait)
+        nodes_file.skip<std::uint64_t>(1); // node_count
+        const auto coordinates_ptr = layout_ptr->GetBlockPtr<util::Coordinate, true>(
+            memory_ptr, DataLayout::COORDINATE_LIST);
+        const auto osmnodeid_ptr =
+            layout_ptr->GetBlockPtr<std::uint64_t, true>(memory_ptr, DataLayout::OSM_NODE_ID_LIST);
+        util::PackedVector<OSMNodeID, true> osmnodeid_list;
+
+        osmnodeid_list.reset(osmnodeid_ptr, layout_ptr->num_entries[DataLayout::OSM_NODE_ID_LIST]);
+
+        io::readNodes(nodes_file,
+                      coordinates_ptr,
+                      osmnodeid_list,
+                      layout_ptr->num_entries[DataLayout::COORDINATE_LIST]);
     }
 
-    if (shared_layout_ptr->GetBlockSize(SharedDataLayout::BEARING_OFFSETS) > 0)
+    // store timestamp
     {
-        auto *bearing_offsets_ptr = shared_layout_ptr->GetBlockPtr<unsigned, true>(
-            shared_memory_ptr, SharedDataLayout::BEARING_OFFSETS);
-        std::copy(bearing_offsets_data.begin(), bearing_offsets_data.end(), bearing_offsets_ptr);
+        io::File timestamp_file(config.timestamp_path);
+        const auto timestamp_size = timestamp_file.size();
+
+        const auto timestamp_ptr =
+            layout_ptr->GetBlockPtr<char, true>(memory_ptr, DataLayout::TIMESTAMP);
+        BOOST_ASSERT(timestamp_size == layout_ptr->num_entries[DataLayout::TIMESTAMP]);
+        timestamp_file.readInto(timestamp_ptr, timestamp_size);
     }
 
-    if (shared_layout_ptr->GetBlockSize(SharedDataLayout::BEARING_BLOCKS) > 0)
+    // store search tree portion of rtree
     {
-        auto *bearing_blocks_ptr =
-            shared_layout_ptr->GetBlockPtr<typename util::RangeTable<16, true>::BlockT, true>(
-                shared_memory_ptr, SharedDataLayout::BEARING_BLOCKS);
-        std::copy(bearing_blocks_data.begin(), bearing_blocks_data.end(), bearing_blocks_ptr);
+        io::File tree_node_file(config.ram_index_path);
+        // perform this read so that we're at the right stream position for the next
+        // read.
+        tree_node_file.skip<std::uint64_t>(1);
+        const auto rtree_ptr =
+            layout_ptr->GetBlockPtr<RTreeNode, true>(memory_ptr, DataLayout::R_SEARCH_TREE);
+        io::readRamIndex(
+            tree_node_file, rtree_ptr, layout_ptr->num_entries[DataLayout::R_SEARCH_TREE]);
     }
 
-    if (!bearing_class_table.empty())
     {
-        auto bearing_class_ptr = shared_layout_ptr->GetBlockPtr<DiscreteBearing, true>(
-            shared_memory_ptr, SharedDataLayout::BEARING_VALUES);
-        std::copy(bearing_class_table.begin(), bearing_class_table.end(), bearing_class_ptr);
+        io::File core_marker_file(config.core_data_path);
+        const auto number_of_core_markers = core_marker_file.readElementCount32();
+
+        // load core markers
+        std::vector<char> unpacked_core_markers(number_of_core_markers);
+        core_marker_file.readInto(unpacked_core_markers.data(), number_of_core_markers);
```

Comment:
would love if we could move away from these `read into the output` with pointers and use proper iterators( e.g. back inserters).